### DISCUSSION TITLE
Why can not `abc = Abc _`?

### DISCUSSION THREAD

**Discussion by anz-ryu** [ocaml/ocaml#12501] [General]
abc is a value of Variant Abc ...

Example :

```ocaml
type Vabc =
  | Abc string
  | Efg

let result = abc = Abc
```

I got `This typed hole should be replaced with an expression of type string`.

but we can do `abc = Efg`.

**Comment by gasche** [comment]
The provided code example is not valid OCaml syntax. My impression is that this issue is missing context, and is probably not best located in the compiler codebase project. If this is a question on the surface OCaml syntax (or a dialect of it?), maybe consider asking on the OCaml [Discuss](https://discuss.ocaml.org/) forum?
